src/InvalidTypeException.java

```diff
@@ -0,0 +1,16 @@
+@SuppressWarnings("serial")
+public class InvalidTypeException extends WordsException {
+	
+	private String expectedType;
+	private String receivedType;
+	
+	public InvalidTypeException(int lineNo, String expected, String received) {
+		super(lineNo);
+		this.expectedType = expected;
+		this.receivedType = received;
+	}
+
+	public String toString() {
```

Comment:
I think it's good practice to @Override the toString() method from Java.object so that one could print the exception without having to explicitly call toString() on it.  (This applies to all the exceptions, just putting the line comment here.)